static/app/requestDeduper.tsx

```diff
@@ -0,0 +1,14 @@
+export default class RequestDeduper<T> {
```

Comment:
Maybe I should pick a better name because since it takes a generic, it can be used for other things than requests

Reply:
Seems fine to me, I don't see any other use cases for a map of strings to promises.